crypto/rand/rand_unix.c

```diff
@@ -256,5 +358,108 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 #  endif
 }
 # endif
+#endif
+
+#ifdef OPENSSL_SYS_UNIX
+int rand_pool_add_nonce_data(RAND_POOL *pool)
+{
+    struct {
+        gid_t curr_gid;
+        pid_t curr_pid;
+        uid_t curr_uid;
+        uint64_t curr_time;
+    } data;
+
+    /*
+     * Add gid, pid, and uid to nonce to ensure variation between
+     * different processes.
```

Comment:
Would it be wise to have the thread ID here as well?  On some systems, that's the only way to differentiate between threads...
(as a matter of fact, wouldn't pid and tid be enough?)